mito-ai/src/components/ModelSelector.tsx

```diff
@@ -0,0 +1,221 @@
+    } else if (isCompact) {setDropdownWidth(28);
+    }
+  }, [selectedModel, isCompact]);
+
+  // Shorten model name if it's too long
+  const truncateModelName = (name: string): string => {
+    if (name.length > MAX_MODEL_NAME_LENGTH) {
+      return `${name.substring(0, MAX_MODEL_NAME_LENGTH - 3)}...`;
+    }
+    return name;
+  };
+
+  const handleModelChange = (model: string) => {
+    setSelectedModel(model);
+    setIsOpen(false);
+
+    const newConfig = {
+      model: model
+    };
+
+    localStorage.setItem('llmModelConfig', JSON.stringify(newConfig));
+
+    // Notify parent component
+    onConfigChange(newConfig);
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const target = event.target as HTMLElement;
+      if (!target.closest('.model-selector')) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (isOpen && dropdownRef.current && containerRef.current) {
+      const dropdown = dropdownRef.current;
+      const container = containerRef.current;
+      const rect = container.getBoundingClientRect();
+
+      if (rect.top < dropdown.offsetHeight) {
+        dropdown.style.bottom = 'auto';
+        dropdown.style.top = '100%';
+        dropdown.style.marginTop = '4px';
+        dropdown.style.marginBottom = '0';
+      } else {
+        // Default: position above
+        dropdown.style.top = 'auto';
+        dropdown.style.bottom = '100%';
+        dropdown.style.marginBottom = '4px';
+        dropdown.style.marginTop = '0';
+      }
+
+      // Ensure it doesn't overflow horizontally
+      if (isCompact) {
+        const leftSpace = rect.left;
+        const rightSpace = window.innerWidth - rect.right;
+
+        if (leftSpace < (dropdown.offsetWidth / 2) || rightSpace < (dropdown.offsetWidth / 2)) {
+          // Not enough space to center, align to the side with more space
+          if (leftSpace < rightSpace) {
+            dropdown.style.left = '0';
+            dropdown.style.right = 'auto';
+            dropdown.style.transform = 'none';
+          } else {
+            dropdown.style.left = 'auto';
+            dropdown.style.right = '0';
+            dropdown.style.transform = 'none';
+          }
+        } else {
+          dropdown.style.left = '50%';
+          dropdown.style.right = 'auto';
+          dropdown.style.transform = 'translateX(-50%)';
+        }
+      }
+    }
+  }, [isOpen, isCompact]);
+
+  return (
+    <div className="model-selector" ref={containerRef}>
+      <div
+        className={`model-selector-dropdown ${isCompact ? 'compact-mode' : ''}`}
+        onClick={() => setIsOpen(!isOpen)}
+        style={{ width: `${dropdownWidth}px` }}
```

Comment:
Instead of hardcoding the `width` here, I think it's better practice to just let the css take care of it. You can set something like `width: min-content` and I think that will have the same impact.